packages/expo-image/android/src/main/java/expo/modules/image/drawing/BorderDrawable.kt

```diff
@@ -0,0 +1,1078 @@
+  private var mBorderWidth: Spacing? = null
+  private var mBorderRGB: Spacing? = null
+  private var mBorderAlpha: Spacing? = null
+  private var mBorderStyle: BorderStyle? = null
+
+  /* Used for rounded border and rounded background */
+  private var mPathEffectForBorderStyle: PathEffect? = null
+  private var mInnerClipPathForBorderRadius: Path? = null
+  private var mOuterClipPathForBorderRadius: Path? = null
+  private var mPathForBorderRadiusOutline: Path? = null
+  private var mPathForBorder: Path? = null
+  private var mCenterDrawPath: Path? = null
+  private var mInnerClipTempRectForBorderRadius: RectF? = null
+  private var mOuterClipTempRectForBorderRadius: RectF? = null
+  private var mTempRectForCenterDrawPath: RectF? = null
+  private var mInnerTopLeftCorner: PointF? = null
+  private var mInnerTopRightCorner: PointF? = null
+  private var mInnerBottomRightCorner: PointF? = null
+  private var mInnerBottomLeftCorner: PointF? = null
+  private var mNeedUpdatePathForBorderRadius = false
+  private var mBorderRadius = YogaConstants.UNDEFINED
+
+  /* Used by all types of background and for drawing borders */
+  private val mPaint = Paint(Paint.ANTI_ALIAS_FLAG)
+  private var mColor = Color.TRANSPARENT
+  private var mAlpha = 255
+  private var mBorderCornerRadii: FloatArray? = null
+
+  /**
+   * Similar to Drawable.getLayoutDirection, but available in APIs < 23.
+   */
+  var resolvedLayoutDirection = 0
+    private set
+
+  enum class BorderRadiusLocation {
+    TOP_LEFT, TOP_RIGHT, BOTTOM_RIGHT, BOTTOM_LEFT, TOP_START, TOP_END, BOTTOM_START, BOTTOM_END
+  }
+
+  override fun draw(canvas: Canvas) {
+    updatePathEffect()
+    if (!hasRoundedBorders() && mBorderStyle == BorderStyle.SOLID) {
+      drawRectangularBackgroundWithBorders(canvas)
+    } else {
+      drawRoundedBackgroundWithBorders(canvas)
+    }
+  }
+
+  private fun hasRoundedBorders(): Boolean {
+    if (!YogaConstants.isUndefined(mBorderRadius) && mBorderRadius > 0) {
+      return true
+    }
+    if (mBorderCornerRadii != null) {
+      for (borderRadii in mBorderCornerRadii!!) {
+        if (!YogaConstants.isUndefined(borderRadii) && borderRadii > 0) {
+          return true
+        }
+      }
+    }
+    return false
+  }
+
+  override fun onBoundsChange(bounds: Rect) {
+    super.onBoundsChange(bounds)
+    mNeedUpdatePathForBorderRadius = true
+  }
+
+  override fun setAlpha(alpha: Int) {
+    if (alpha != mAlpha) {
+      mAlpha = alpha
+      invalidateSelf()
+    }
+  }
+
+  override fun getAlpha(): Int {
+    return mAlpha
+  }
+
+  override fun setColorFilter(cf: ColorFilter?) {
+    // do nothing
+  }
+
+  override fun getOpacity(): Int {
+    return ColorUtil.getOpacityFromColor(ColorUtil.multiplyColorAlpha(mColor, mAlpha))
+  }
+
+  /* Android's elevation implementation requires this to be implemented to know where to draw the shadow. */
+  override fun getOutline(outline: Outline) {
+    if (Build.VERSION.SDK_INT < Build.VERSION_CODES.LOLLIPOP) {
```

Comment:
There are few `if(Build.VERSION.SDK_INT < x)` in this file that Android Studio marks as dead code. What's the minimum version of Android that Expo supports?

Reply:
We currently support API above 21 (Lollipop). So yes, we can remove this if :D 
See:
https://github.com/expo/expo/blob/master/apps/bare-expo/android/build.gradle#L9